src/ChaiVM/interpreter/code-manager.cpp

```diff
@@ -15,17 +18,54 @@ void CodeManager::load(std::istream &istream) {
     }
 }
 
+void CodeManager::loadPool(std::istream &istream) {
+    if (!istream.good()) {
+        throw std::invalid_argument(std::string{"Bad input stream"});
+    }
+    uint16_t constants = 0;
+    static_assert(sizeof constants == 2);
```

Comment:
@c71n93 let's leave it. What if we decide write number of constants to file in 4 bytes instead of 2, but immediate will still be 2 bytes. I leave this assert just for safety